pkg/apis/core/v1/conversion.go

```diff
@@ -270,6 +274,40 @@ func Convert_v1_PodTemplateSpec_To_core_PodTemplateSpec(in *v1.PodTemplateSpec,
 	return nil
 }
 
+func Convert_v1_PodStatus_To_core_PodStatus(in *v1.PodStatus, out *core.PodStatus, s conversion.Scope) error {
+	if err := autoConvert_v1_PodStatus_To_core_PodStatus(in, out, s); err != nil {
+		return err
+	}
+
+	// If both fields (v1.PodIPs and v1.PodIP) are provided, then test v1.PodIP == v1.PodIPs[0]
+	if (len(in.PodIP) > 0 && len(in.PodIPs) > 0) && (in.PodIP != in.PodIPs[0].IP) {
+		return fmt.Errorf("conversion Error: in.PodIP(%v) != in.PodIPs[0](%v)", in.PodIP, in.PodIPs[0].IP)
```

Comment:
this will surface to API consumers, to whom `in` will have no meaning. suggest using the serialized field names `status.podIP` and `status.podIPs[0].ip` instead

Reply:
actually, won't kubelets < 1.16 encounter this if they attempt to change a pod's IP?

1. v1.15 kubelet sets pod status.podIP=1.2.3.4 running against a 1.16 API server
2. v1.16 API server populates status.podIPs={ip:1.2.3.4} during conversion
3. v1.15 kubelet patches pod status.podIP to 2.3.4.5 to change the IP, preserving existing content of unknown status.podIPs field
4. v1.16 API server returns a conversion error because status.podIP != status.podIPs[0]

If `status.podIP` and `status.podIPs` are both set and `status.podIP != status.podIPs[0]`, don't we have to honor  `status.podIP`? This is the discussion we had in https://groups.google.com/d/topic/kubernetes-sig-architecture/4Ibjx2pn1aU/discussion, specifically https://groups.google.com/d/msg/kubernetes-sig-architecture/4Ibjx2pn1aU/AvShSt3iAQAJ

Reply:
I'm not very familiar with the kubelet bits that set status.podIP... does it ever try to change a podIP? If not, erroring here may happen to be safe for the single writer we know about, but is not a great approach in general

Reply:
Kubelet does not change `podIP` except from ` "" => IP`. Also considering the case above.  3rd item above kubelet patches - if it ever happened, which AFIAK it does not, but should it -, it will look like this `PodIP=<new IP> PodIPs=nil`, which will be a valid call and will convert as expected.

Reply:
> 3rd item above if kubelet patches PodIP/s then it will look like this PodIP=<new IP>,PodIPs=nil

A 1.15 kubelet would not be aware of the PodIPs field, so it would just send a patch to change the PodIP field, right? Wouldn't that result in both PodIP and PodIPs being set?

Reply:
(if kubelet never changes PodIP it may be a moot point… are we positive it never attempts to do that?)

Reply:
According to what i have seen in the code, no.